store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreTest.java

```diff
@@ -255,12 +256,22 @@ public void shouldNotAllowOverwritingOfGraphWithinFederatedScope() throws Except
             assertContains(e, "GraphId: ", ACC_ID_2);
         }
     }
-
-    @Test(expected = UnsupportedOperationException.class)
-    public void shouldDoUnhandledOperation() throws Exception {
-        store.doUnhandledOperation(null, null);
+    @Test
+    public void shouldThrowAppropriateExceptionWhenHandlingAnUnsupportedOperation() {
+        // Given
+        Operation op = new OperationImpl();
+        // When
+        try {
+            store.handleOperation(op, new Context());
+            fail("Exception expected");
+        } catch (final OperationException e) {
+            fail("Expected an UnsupportedOperationException rather than an OperationException");
+        } catch (final UnsupportedOperationException e) {
+            // Then
+            assertEquals("Operation class uk.gov.gchq.gaffer.operation.impl.OperationImpl is not supported by the FederatedStore.", e.getMessage());
+        }
     }
-
+    
```

Comment:
Thanks for adding that. It compiles perfectly. Only reason checkstyle is failing is because this line has trailing spaces. Removing this should fix it
```suggestion

```